server/src/internalClusterTest/java/org/elasticsearch/operateAllIndices/DestructiveOperationsIT.java

```diff
@@ -155,4 +155,40 @@ public void testAddIndexBlockDefaultBehaviour() throws Exception {
         assertTrue("write block is set on index1", state.getBlocks().hasIndexBlock("index1", IndexMetadata.INDEX_WRITE_BLOCK));
         assertTrue("write block is set on 1index", state.getBlocks().hasIndexBlock("1index", IndexMetadata.INDEX_WRITE_BLOCK));
     }
+
+    public void testRemoveIndexBlockIsRejected() throws Exception {
+        updateClusterSettings(Settings.builder().put(DestructiveOperations.REQUIRES_NAME_SETTING.getKey(), true));
+
+        createIndex("index1", "1index");
+        // Add blocks first
+        assertAcked(indicesAdmin().prepareAddBlock(WRITE, "index1", "1index").get());
+
+        // Should succeed, since no wildcards
+        assertAcked(indicesAdmin().prepareRemoveBlock(WRITE, "1index").get());
+        // Special "match none" pattern succeeds, since non-destructive
+        assertAcked(indicesAdmin().prepareRemoveBlock(WRITE, "*", "-*").get());
+
+        expectThrows(IllegalArgumentException.class, indicesAdmin().prepareRemoveBlock(WRITE, "i*"));
+        expectThrows(IllegalArgumentException.class, indicesAdmin().prepareRemoveBlock(WRITE, "_all"));
```

Comment:
Can you switch the order - i.e. first do the rejected requests? Otherwise those requests don't really make sense (on non existing indices).